NEW_README.md

```diff
@@ -141,15 +140,155 @@ Run prints its output if either,
+
+```toml
+[[task]]
+  id = "dev"
+  type = "long"
+  dependencies = ["simulate-coding"]
+  triggers = ["build-css", "build-js"]
+  watch = ["server-config.json"]
+  env = {
+    KEY = "some value"
+    OTHER_KEY = "some other value"
+  }
+  cmd = """
+    echo "dev-server running at http://localhost:3000"
+    while true; do sleep 1; done
+  """
+```
+
+Explore a [realistic project configuration](https://github.com/amonks/run/tree/main/examples/demo) to see `tasks.toml` in action.
+
+### Understanding Task Fields
+
+#### Required Fields
+
+Task definitions have two required fields: `id` and `type`. _Note_: if only the required fields are set and none of `dependencies`, `triggers`, or `cmd` are specified, then the task is a no-op.
+
+##### `id`
+
+The unique identifier for a task. It is used for:
+- Command line invocation (`$ run <id>`)
+- Identifying the task in the interactive UI (TUI).
+
+##### `type`
+
+Specifies the task's lifecycle management strategy:
+- A task can be of only one type: `long` or `short`, any other value is invalid
+- **`long`**: The task is kept alive indefinitely. It's restarted if it exits unexpectedly. **Not suitable as a trigger**.
+  - For example, a development server or a test runner.
+  - If a task depends on a "long" task, Run doesn't really know when the long task has produced whatever output is depended on, so the dependent is run 500ms after the long task starts.
+- **`short`**: The task is considered complete upon successful execution. If it fails, Run will retry it after a short delay.
+  - For example, a build script or a code generation task.
+
+#### Optional Fields
+
+##### `description`
+
+A human-readable description of the task. It's displayed in the TUI task list, the output of `run -list`, and can be used to document the task's purpose. It can be one line or multiline.
+
+##### `dependencies`
+Lists other task IDs that should run alongside this task. 
+
+- If task A depends on task B, B starts before A.
+- If B is a "long" task, A will start 500ms after B starts, and if B is a "short" task, A will start as soon as B completes.[^](#type)
+  
+##### `triggers`
+Lists other task IDs that should run alongside this task, and when completed successfully, cause this task to restart. 
+
+- Triggers are similar to dependencies but specifically for re-execution upon successful completion of the trigger task.
+- If task A is triggered by task B, and both A and B are running, a successful completion of B will trigger A to restart.
+- **Triggers are not considered when a task is initially run.** They only affect task restarts.
+- **Long tasks cannot be triggers.** It is invalid to use a long task as a trigger, since long tasks aren't expected to end.
+
+>[!NOTE]
+> Task IDs listed as `dependencies` and `triggers` can cross directory boundaries (e.g., `"css/build"` refers to a task in `./css/tasks.toml`).
```

Comment:
todo later: "cross directory boundaries" has some caveats; might be good to add a separate "modular taskfiles" section describing this feature and its quirks in detail